src/Link.php

```diff
@@ -7,4 +7,44 @@
  */
 class Link extends Text
 {
+
+    public function __construct($text = "", $href = "", $attributes = [])
+    {
+        parent::__construct($text);
+        $this->href = $href;
+        $this->attributes = $attributes;
+    }
+
+    public function write()
+    {
+        if ($this->text == "") return self::beautify($this->text);
+
+        if (preg_match("/\n/", $this->text)) {
+            $tags = explode("\n", $this->text);
+            $tags = array_map(function($tag) {
+              if (preg_match(blockRegex, $tag)) return $tag;
+              if (!trim($tag)) return $tag;
```

Comment:
FYI we could use `empty(trim($tag))` if it reads better. I think this is fine as-is.